database-restore-postgres.sh

```
@@ -0,0 +1,289 @@
+    fi
+    
+    if [[ -z "$sql_file" ]]; then
+        log "WARNING: No backup file found for schema '$schema'"
+        ((failure_count++))
+        continue
+    fi
+    
+    # Restore the schema
+    restore_schema "$schema" "$sql_file"
+    
+    if [[ $? -eq 0 ]]; then
```

Shellcheck (0.10.0)

[style] Check exit code directly with e.g. 'if mycmd;', not indirectly with $?.

(SC2181)